Add FindStaffCommand

AugustDespair opened the pull request:
Added the ability to find a staff in a specified project.

Syntax: `findstaff pn/Project Name sn/Staff Name`

Potential To-Dos:
- Perhaps we can extend to other Staff fields?
- More extensive test cases if I can think of any 

Closes #86, #95

maxng17 commented on src/main/java/seedu/address/logic/commands/FindStaffCommand.java:
```diff
@@ -0,0 +1,102 @@
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Project> lastShownList = model.getFilteredProjectList();
```
Now that I think about it, I not sure if this follow OOP principal by getting model fields. If it violates we might have to look through the past commands that we make and change them.

maxng17 commented on src/main/java/seedu/address/logic/commands/FindStaffCommand.java:
```diff
@@ -0,0 +1,102 @@
+
+        for (Staff staff : projectToFind.getStaffList()) {
+            if (keywords.stream()
+                    .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(staff.getStaffName().toString(), keyword))) {
```
I think we have the class StaffNameContainsKeywordPredicate in the staff folder. Their original implementation for find uses such a class to set a predicate for the filter list. Maybe we should replicate their design since setting a predicate will update the filteredList that updates the UI

maxng17 commented on src/test/java/seedu/address/logic/commands/AddStaffCommandTest.java:
```diff
@@ -37,14 +37,14 @@ public void execute_validProjectValidStaff_success() {
         AddStaffCommand addStaffCommand = new AddStaffCommand(validStaff, validProject.getProjectName());
 
-        String expectedMessaage = String.format(AddStaffCommand.MESSAGE_ADD_STAFF_SUCCESS, validStaff);
+        String expectedMessage = String.format(AddStaffCommand.MESSAGE_ADD_STAFF_SUCCESS, validStaff);
```
Nice catch on typo

maxng17 reviewed (commented):
I think the test case for FindStaffCommandParser is not included yet, also need to see if we want to follow the current implementation for FindStaffComamnd. Apart from that the rest looks good.

AugustDespair commented:
Updated the `FindStaffCommand` to find the current active Project.

Some auxillary changes that followed:
- changed some variable namings inside `StaffBuilder` to be more descriptive
- `viewCommand()`: I added `model.updateFilteredStaffList(Model.PREDICATE_SHOW_ALL_STAFF)` because without this, calling `view` would still show the filtered staff list instead of all staff.
- added `containsNameIgnoreCase` to `StringUtil` to facilitate checking if a `keyword` is inside a `name`, ignoring cases.

jjtan444 reviewed (approved):
Overall high quality PR with substantial test cases. Well done. LGTM
